<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>文本编辑器</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="CodeEditor" name="plainTextEdit"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <widget class="QMenu" name="menuFile">
    <property name="title"><string>文件(&amp;F)</string></property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title"><string>编辑(&amp;E)</string></property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionSelectAll"/>
    <addaction name="separator"/>
    <addaction name="actionFind"/>
    <addaction name="actionReplace"/>
   </widget>
  <widget class="QMenu" name="menuView">
    <property name="title"><string>视图(&amp;V)</string></property>
    <addaction name="actionToggleToolbar"/>
    <addaction name="actionToggleStatusBar"/>
  </widget>
  <widget class="QMenu" name="menuFormat">
    <property name="title"><string>格式(&amp;O)</string></property>
    <addaction name="actionToggleWordWrap"/>
    <addaction name="actionToggleLineNumbers"/>
    <addaction name="actionFont"/>
    <addaction name="actionTextColor"/>
    <addaction name="actionTextBackgroundColor"/>
    <addaction name="actionBackgroundColor"/>
  </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title"><string>帮助(&amp;H)</string></property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuFormat"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle"><string>工具栏</string></property>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionCut"/>
   <addaction name="actionCopy"/>
   <addaction name="actionPaste"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="actionFind"/>
   <addaction name="actionReplace"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNewWindow">
   <property name="text"><string>新建窗口</string></property>
   <property name="shortcut"><string>Ctrl+Shift+N</string></property>
  </action>
  <action name="actionNew">
   <property name="text"><string>新建</string></property>
   <property name="shortcut"><string>Ctrl+N</string></property>
  </action>
  <action name="actionOpen">
   <property name="text"><string>打开...</string></property>
   <property name="shortcut"><string>Ctrl+O</string></property>
  </action>
  <action name="actionSave">
   <property name="text"><string>保存</string></property>
   <property name="shortcut"><string>Ctrl+S</string></property>
  </action>
  <action name="actionSaveAs">
   <property name="text"><string>另存为...</string></property>
   <property name="shortcut"><string>Ctrl+Shift+S</string></property>
  </action>
  <action name="actionPrint">
   <property name="text"><string>打印...</string></property>
   <property name="shortcut"><string>Ctrl+P</string></property>
  </action>
  <action name="actionExit">
   <property name="text"><string>退出</string></property>
   <property name="shortcut"><string>Ctrl+Q</string></property>
  </action>
  <action name="actionUndo">
   <property name="text"><string>撤销</string></property>
   <property name="shortcut"><string>Ctrl+Z</string></property>
  </action>
  <action name="actionRedo">
   <property name="text"><string>恢复</string></property>
   <property name="shortcut"><string>Ctrl+Shift+Z</string></property>
  </action>
  <action name="actionCut">
   <property name="text"><string>剪切</string></property>
   <property name="shortcut"><string>Ctrl+X</string></property>
   <property name="enabled"><bool>false</bool></property>
  </action>
  <action name="actionCopy">
   <property name="text"><string>复制</string></property>
   <property name="shortcut"><string>Ctrl+C</string></property>
   <property name="enabled"><bool>false</bool></property>
  </action>
  <action name="actionPaste">
   <property name="text"><string>粘贴</string></property>
   <property name="shortcut"><string>Ctrl+V</string></property>
   <property name="enabled"><bool>false</bool></property>
  </action>
  <action name="actionSelectAll">
   <property name="text"><string>全选</string></property>
   <property name="shortcut"><string>Ctrl+A</string></property>
  </action>
  <action name="actionFind">
   <property name="text"><string>查找...</string></property>
   <property name="shortcut"><string>Ctrl+F</string></property>
  </action>
  <action name="actionReplace">
   <property name="text"><string>替换...</string></property>
   <property name="shortcut"><string>Ctrl+H</string></property>
  </action>
  <action name="actionFont">
   <property name="text"><string>字体...</string></property>
  </action>
  <action name="actionTextColor">
   <property name="text"><string>字体颜色...</string></property>
  </action>
  <action name="actionBackgroundColor">
    <property name="text"><string>编辑器背景色...</string></property>
  </action>
  <action name="actionTextBackgroundColor">
   <property name="text"><string>字体背景色...</string></property>
  </action>
  <action name="actionToggleToolbar">
   <property name="text"><string>显示工具栏</string></property>
   <property name="checkable"><bool>true</bool></property>
   <property name="checked"><bool>true</bool></property>
  </action>
  <action name="actionToggleStatusBar">
   <property name="text"><string>显示状态栏</string></property>
   <property name="checkable"><bool>true</bool></property>
   <property name="checked"><bool>true</bool></property>
  </action>
  <action name="actionToggleLineNumbers">
   <property name="text"><string>显示行号</string></property>
   <property name="checkable"><bool>true</bool></property>
   <property name="checked"><bool>true</bool></property>
  </action>
  <action name="actionToggleWordWrap">
   <property name="text"><string>自动换行</string></property>
   <property name="checkable"><bool>true</bool></property>
   <property name="checked"><bool>true</bool></property>
  </action>
  <action name="actionAbout">
   <property name="text"><string>关于</string></property>
  </action>
  <action name="actionAutoSave">
   <property name="text"><string>自动保存</string></property>
   <property name="checkable"><bool>true</bool></property>
   <property name="checked"><bool>false</bool></property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <connections/>
</ui>